### ISSUE TITLE
http-server

### ISSUE THREAD

**Issue by ptrepanier** [microsoft/vscode#161093]
I'm not sure if this is a github issue or not, but within codespace, we use a command http-server.   It seems to start ok, but when I visit the link I always get a 404 error such as

{"type":"https://tools.ietf.org/html/rfc7231#section-6.5.4","title":"Not Found","status":404,"traceId":"00-49cf1378947ca136e793a4efec5fa476-0b27de5d55902dcd-00"}

http-server says that the site can be reached at this url
https://ptrepanier-code50-33074330-7wqv9pvrfp679-8080.githubpreview.dev

is this a github issue or a CS50 issue?

Version: 1.71.2
Commit: 74b1f979648cc44d385a2286793c226e611f59e7
User Agent: Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/105.0.0.0 Safari/537.36
Embedder: codespaces

**Comment by ptrepanier** [comment]
Hi.  I moved onto week9 of CS50 using flask.  I have the same type of problem.  Flask starts correctly in the correct subfolder that has app.py but my URL does not return the expected results:

e.g when I type 
$> flask run the server starts ok but the response is not correct 
INFO: WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
 * Running on all addresses (0.0.0.0)
 * Running on https://ptrepanier-code50-33074330-7wqv9pvrfp679-5000.githubpreview.dev
 * Running on https://ptrepanier-code50-33074330-7wqv9pvrfp679-5000.githubpreview.dev
INFO: Press CTRL+C to quit

Url becomes 
https://ptrepanier-code50-33074330-7wqv9pvrfp679-5000.githubpreview.dev/authenticate-codespace/ptrepanier-code50-33074330-7wqv9pvrfp679?cid=dc5a599c9593601a3b7651929600e5e0&port=5000&visibility=private

and response is:

{"type":"https://tools.ietf.org/html/rfc7231#section-6.5.1","title":"Bad Request","status":400,"traceId":"00-1870f3b21709196680fef9877b92ada3-1c7188949b9fab85-00"}

Sometimes I get a 400 error and sometimes a 404.

The strange part of this is it did work for a while a few days ago.  I was using http-server and it did return the correct directory listing, and when I clicked on my static index.html it worked fine.    Then, a few hours later, nothing was working and I thought it might just go away (server issue) but now it's been several days and I really need flask to work.

**Comment by ptrepanier** [comment]
I got a response from harvard sysadmins and they suggested I create a new codespace which I did and now it works.  For those of us new to codespaces that is not an obvious solution.  I had to copy some files from my old codespace to the new one but at least the new one is working and I can go into both codespaces to compare and copy files (that were missing in the new codespace).  The "sync" (which I don't know how to do or if it's automatic) was not working with my old codespace.
